src/frontend/src/icp-eth/components/tokens/AddTokenByNetwork.svelte

```diff
@@ -0,0 +1,40 @@
+<script lang="ts">
+	import { i18n } from '$lib/stores/i18n.store';
+	import { Dropdown, DropdownItem } from '@dfinity/gix-components';
+	import { networksMainnets } from '$lib/derived/networks.derived';
+	import { ETHEREUM_NETWORK_SYMBOL, ICP_NETWORK_SYMBOL } from '$env/networks.env';
+	import IcAddTokenForm from '$icp/components/tokens/IcAddTokenForm.svelte';
+	import AddTokenForm from '$eth/components/tokens/AddTokenForm.svelte';
+	import { networkId } from '$lib/derived/network.derived';
+	import { isNullish } from '@dfinity/utils';
+
+	export let dropdownNetwork: string | undefined;
+	export let tokenData: Record<string, string>;
+
+	let ledgerCanisterId: string = '';
```

Comment:
There shouldn't be a callback that reset the variable if the network is switched or UX wise we want to keep the previous values in memories when the user select networks back and forth?

Reply:
Works for me in terms of UX, I'm just a bit concern about the state because the parent/modal component will ends up knowing about values that should not be persisted.

e.g. user select IC, enter ledger ID, then select Ethereum, enter contract address, finish the process and when the `save` is called, there is a valid ledger ID in memory.

But I guess that if we really pay attention...

Reply:
if you want (in another PR) we can improve this doing something like modifying slightly the icNext event in `AddTokenByNetwork`, for example

```svelte
const icNext = () => {
  contractAddress = '';
  dispatch('icNext');
  }

<IcAddTokenForm on:icBack on:icNext={icNext} bind:ledgerCanisterId bind:indexCanisterId />
```

and same logic for `AddTokenForm`

Reply:
something like that or a clean onDestroy of the component would be nice indeed

Reply:
in another pr sure

Reply:
created PR #1507 for this